compiler/plugins/target/AMD-AIE/aie/AMDAIECreatePathFindFlows.cpp

```diff
@@ -573,22 +158,25 @@ LogicalResult DynamicTileAnalysis::runAnalysis(DeviceOp &device) {
     TileOp dstTile = cast<TileOp>(flowOp.getDest().getDefiningOp());
     TileLoc srcCoords = {srcTile.colIndex(), srcTile.rowIndex()};
     TileLoc dstCoords = {dstTile.colIndex(), dstTile.rowIndex()};
-    Port srcPort = {flowOp.getSourceBundle(), flowOp.getSourceChannel()};
-    Port dstPort = {flowOp.getDestBundle(), flowOp.getDestChannel()};
-    LLVM_DEBUG(llvm::dbgs()
-               << "\tAdding Flow: (" << srcCoords.col << ", " << srcCoords.row
-               << ")" << stringifyWireBundle(srcPort.bundle) << srcPort.channel
-               << " -> (" << dstCoords.col << ", " << dstCoords.row << ")"
-               << stringifyWireBundle(dstPort.bundle) << dstPort.channel
-               << "\n");
+    Port srcPort = {toStrmT(flowOp.getSourceBundle()),
+                    flowOp.getSourceChannel()};
+    Port dstPort = {toStrmT(flowOp.getDestBundle()), flowOp.getDestChannel()};
     pathfinder->addFlow(srcCoords, srcPort, dstCoords, dstPort);
   }
 
   // add existing connections so Pathfinder knows which resources are
   // available search all existing SwitchBoxOps for exising connections
   for (SwitchboxOp switchboxOp : device.getOps<SwitchboxOp>()) {
     for (ConnectOp connectOp : switchboxOp.getOps<ConnectOp>()) {
-      if (!pathfinder->addFixedConnection(connectOp))
+      auto sb = connectOp->getParentOfType<SwitchboxOp>();
+      // TODO: keep track of capacity?
+      if (sb.getTileOp().isShimNOCTile()) continue;
```

Comment:
Could you elaborate? What needs to happen for Shim tiles?

Reply:
The next line tries to add an existing (i.e., hand-written) connection and errors if that fails. I don't know/remember why shim noc tiles are/were treated differently (they're given special treatment in various places for unclear reasons) but the TODO was the thought that even if we don't try to add those connections (as actual routes) we might consider checking if capacity has been exceeded.  

Regardless this doesn't apply to us because we never handwrite anything (so the loop could be deleted) but you might at some point want to (in order to debug?). Figuring out why/how shim nocs are treated differently I'll leave to this [follow-up](https://github.com/nod-ai/iree-amd-aie/issues/540) that I wasn't planning but is now necessary since we want to use control packets.